CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
-cmake_minimum_required(VERSION 2.8)
-
-project(FlatBuffers)
+cmake_minimum_required(VERSION 3.0)
+
+# Infer version from git describe
+EXECUTE_PROCESS(
```

Comment:
The main reason for the CMake version bump is for the generator script that creates `<install_prefix>/lib/cmake/flatbuffers/FlatBuffersConfigVersion.cmake`. This script also needs the package version, which is why it is extracted out of Git. This is pretty much based on how it is done in the deb creator that was already in the codebase